gradio/route_utils.py

```diff
@@ -700,6 +700,33 @@ def update_root_in_config(config: BlocksConfigDict, root: str) -> BlocksConfigDi
     return config
 
 
+def update_root_in_api_info(api_info: dict[str, Any], root: str) -> dict[str, Any]:
+    """
+    Updates the root url in the api_info dictionary to the new root url.
+    """
+
+    def _add_root_url(file_dict: dict):
+        default_value = file_dict.get("parameter_default")
+        if default_value is not None and client_utils.is_file_obj_with_url(
+            default_value
+        ):
+            if client_utils.is_http_url_like(default_value["url"]):
+                return file_dict
+            # If running locally or in an insecure evironment use the publicly accessible example_input
+            # to avoid SSRF checks
+            if root.startswith("https"):
```

Comment:
On spaces you'll get the correct default value url. Example here: https://huggingface.co/spaces/freddyaboulton/image-mod-5.0-test

Reply:
This is nice @freddyaboulton but I think we should just use the default value everywhere because:

(1) This is not totally SSRF-proof. You can have localhost URLs that begin with `https` if you have a local ssl certificate. So this could potentially return a `https://localhost...` url that will fail to go through our SSRF validation
(2) The data returned by `/info` can be quite long so this root url replacement can add a delay in how long it takes for the view api page to open for complex gradio apps
(3) Honestly, the `/file=` urls are kind of long and ugly and result in a lot of horizontal scrolling. I prefer the simple github urls

<img width="963" alt="image" src="https://github.com/user-attachments/assets/68d5f61f-824f-462f-852a-b47a0e103ae9">

Reply:
i.e. let's just remove the the logic of setting the `/info` examples values to the be the initial value of the component

Reply:
Ok sounds good. I just wanted to preserve the existing behavior as much as possible.